rest/data/project.go

```diff
@@ -80,6 +80,81 @@ func (pc *DBProjectConnector) UpdateRepo(repoRef *model.RepoRef) error {
 	return nil
 }
 
+// SaveProjectSettingsForSection passes in the existing state of the project page, but saves only the pieces
+// related to the specified section. This handles project ref, webhooks, auth, and project variables.
+func (pc *DBProjectConnector) SaveProjectSettingsForSection(ctx context.Context, projectId string, changes *restModel.APIProjectSettings,
+	section model.ProjectRefSection, userId string) error {
+	before, err := model.GetProjectSettingsById(projectId)
+	if err != nil {
+		return errors.Wrap(err, "error getting before project settings event")
+	}
+	v, err := changes.ProjectRef.ToService()
+	if err != nil {
+		return errors.Wrap(err, "error converting project ref")
+	}
+	newProjectRef := v.(*model.ProjectRef)
+	// if the project ref doesn't use the repo, then this will just be the same as newProjectRef
+	// used to verify that if something is set to nil in the request, we properly validate using the merged project ref
+	mergedProjectRef, err := model.GetProjectRefMergedWithRepo(*newProjectRef)
+	if err != nil {
+		return errors.Wrapf(err, "error merging project ref")
+	}
+
+	catcher := grip.NewBasicCatcher()
+	modified := false
+	switch section {
+	// aliases and notifications use a different connector and should be handled separated
+	case model.ProjectRefGeneralSection:
+		// check if webhook is enabled if the owner/repo has changed
+		if mergedProjectRef.Owner != before.ProjectRef.Owner || mergedProjectRef.Repo != before.ProjectRef.Repo {
+			_, err = pc.EnableWebhooks(ctx, mergedProjectRef)
+			if err != nil {
+				return errors.Wrapf(err, "Error enabling webhooks for project '%s'", projectId)
+			}
+		}
+	case model.ProjectRefAccessSection:
+		// For any admins that are only in the original settings, remove access.
+		// For any admins that are only in the updated settings, give them access.
+		adminsToDelete, adminsToAdd := utility.StringSliceSymmetricDifference(before.ProjectRef.Admins, mergedProjectRef.Admins)
+		if err = pc.UpdateAdminRoles(newProjectRef, adminsToAdd, adminsToDelete); err != nil {
+			return errors.Wrap(err, "error updating admin roles")
+		}
+		modified = true
+		if !before.ProjectRef.IsRestricted() && mergedProjectRef.IsRestricted() {
+			catcher.Wrap(before.ProjectRef.MakeRestricted(), "error making project restricted")
+		}
+		if before.ProjectRef.IsRestricted() && !mergedProjectRef.IsRestricted() {
+			catcher.Wrap(before.ProjectRef.MakeUnrestricted(), "error making unrestricted")
+		}
+
+	case model.ProjectRefVariablesSection:
+		// Remove any variables that only exist in the original settings. (It's possible we can send this information
```

Comment:
what do you mean by "(It's possible we can send this information from the UI in the future instead)"?

Reply:
Here we're building toDelete based on what existed before vs what exists now. We could just build toDelete by sending that information over from the frontend (i.e. adding the variable key to toDelete if the delete button is pressed)

Reply:
ah I see. I agree with the approach of putting more in the backend, and re-assessing if we need less when building out the UI.